modules/codegen/src/smithy4s/codegen/internals/Renderer.scala

```diff
@@ -164,10 +164,42 @@ private[internals] class Renderer(compilationUnit: CompilationUnit) { self =>
       }
   }
 
+  /**
+    * Returns the given list of Smithy documentation strings formatted as Scaladoc comments
+    *
+    * @param l splitted docstring from the IR
+    * @return formatted list of scaladoc lines
+    */
+  private def makeDocLines(l: List[String]): List[Line] = {
+    @tailrec
+    def loop(l: List[String], acc: List[Line]): List[Line] = {
+      l match {
+        case Nil => acc
+        case hd :: tl if tl == Nil =>
+          loop(tl, acc :+ line"  * $hd" :+ line"  */")
+        case hd :: tl => loop(tl, acc :+ line"  * $hd")
+      }
+    }
+    loop(l.tail, List(line"/** ${l.head}"))
```

Comment:
we assume the list is non-empty, shall we encode this into types? Cats is available in the codegen module AFAIK.

Reply:
I have very very superficial experience with Cats:
```scala
  private def documentationAnnotation(hints: List[Hint]): Lines = {
    hints
      .collectFirst { case h: Hint.Documentation => h }
      .flatMap { doc =>
        NonEmptyList.fromList(doc.docString.linesIterator.toList)
      }
      .foldMap(nel => {
        val lines =
          if (nel.size > 1) makeDocLines(nel)
          else List(line"/** ${nel.head} */")
        Lines(lines)
      })
  }
```
Something like this?